js/ui/grid_core/ui.grid_core.rows.js

```diff
@@ -314,72 +314,82 @@ export const rowsModule = {
+                            const rowInsertedClass = this.addWidgetPrefix(ROW_INSERTED_ANIMATION_CLASS);
+
+                            switch(changeType) {
+                                case 'update':
+                                    each(change.rowIndices, (index, rowIndex) => {
+                                        const $newRowElement = this._getRowElements(newTableElement).eq(index);
+                                        const changeType = change.changeTypes && change.changeTypes[index];
+                                        const item = change.items && change.items[index];
+
+                                        executors.push(() => {
+                                            const $rowsElement = this._getRowElements(tableElement);
+                                            const $rowElement = $rowsElement.eq(rowIndex);
+
+                                            switch(changeType) {
+                                                case 'update':
+                                                    if(item) {
+                                                        const columnIndices = change.columnIndices && change.columnIndices[index];
+                                                        if(isDefined(item.visible) && item.visible !== $rowElement.is(':visible')) {
+                                                            $rowElement.toggle(item.visible);
+                                                        } else if(columnIndices) {
+                                                            this._updateCells($rowElement, $newRowElement, columnIndices);
+                                                        } else {
+                                                            $rowElement.replaceWith($newRowElement);
+                                                        }
+                                                    }
+                                                    break;
+                                                case 'insert':
+                                                    if(!$rowsElement.length) {
+                                                        if(tableElement) {
+                                                            const target = $newRowElement.is('tbody') ? tableElement : tableElement.children('tbody');
+                                                            $newRowElement.prependTo(target);
+                                                        }
+                                                    } else if($rowElement.length) {
+                                                        $newRowElement.insertBefore($rowElement);
                                                     } else {
-                                                        $rowElement.replaceWith($newRowElement);
+                                                        $newRowElement.insertAfter($rowsElement.last());
```

Comment:
Is that code reachable?

Because we already handle ```$rowElement.length``` and ```!$rowElement.length``` before this else block

Reply:
!$row**s**Element.length and $rowElement.length - different variables are checked here

Reply:
I think it's better to rename the $row**s**Element variable to $rowElement**s**